### DISCUSSION TITLE
Feature request: Improve 3D visualization tool egs_view (Qt5)

### DISCUSSION THREAD

**Discussion by rtownson** [nrc-cnrc/EGSnrc#693] [Ideas]
If anyone is interested in tackling these issues but is unfamiliar with EGSnrc or compiling `egs_view`, feel free to ask for help on how to get started. The code for `egs_view` is found in `HEN_HOUSE/egs++/view`. The environment variable QTDIR must be set to the directory containing the QT bin (e.g. `~/Qt5/5.10.1/gcc_64`) in order to compile with `make` in the `view` directory.

**Free look mode**
Currently there is a fairly restrictive view mode with rotation around a fixed point in the geometry. If the focus point is allowed to go outside the geometry, the rendering is not robust. Free look mode would be a nice improvement. It is important to maintain the ability to zoom in to view very small details, or zoom out to view a very large geometry without distortion.

**Efficiency boosts**
All rendering is done on the CPU to make use of our specialized geometry routines (`howfar()`) that are also used for particle transport in EGSnrc, so this limits the use of rendering libraries. Any efficiency boosts would be great, such as improved threading or better particle track rendering. When a large number of particle tracks (.ptrack files generated from certain simulations) are loaded, the viewer may slow down. Geometries with many (>100k) regions also slow down.

**Comment by ftessier** [comment]
@rtownson, can you add these to #790, under a new heading section for `egs_view`?
